core/common/src/main/java/alluxio/conf/PrefixPathMatcher.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.conf;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Matches the path with the longest matched prefix, and the path must be in the subtree of the
+ * prefix.
+ *
+ * For example:
+ * /a/b/c matches /a/b, but
+ * /a/bc does not match /a/b because although /a/b is a prefix of /a/bc, /a/bc is not
+ * in the subtree of /a/b.
+ *
+ * Thread safety:
+ * Must be constructed in a single thread and published safely.
+ * After construction, all operations on the internal read only data structures are thread safe.
+ */
+@ThreadSafe
+public final class PrefixPathMatcher implements PathMatcher {
+  /**
+   * A node in a trie.
+   */
+  private final class TrieNode {
+    private Map<Character, TrieNode> mChildren = new HashMap<>();
```

Comment:
Can we store `String` components instead of `Character`s? Then we only need 1 node per component instead of 1 node per character. It will also save a lot of map lookups when traversing